c2corg_ui/externs/appx.js

```diff
@@ -84,3 +84,186 @@ appx.BootstrapSliderValues;
  * }}
  */
 appx.auth.RequestChangePassword;
+
+
+/**
+ * @typedef {{
+ *   routes : Array<Object>,
+ *   waypoints : Array<Object>,
+ *   images : Array<Object>,
+ *   users : Array<Object>,
+ *   recent_outings : Array<Object>,
+ * }}
+ */
+appx.DocumentAssociations;
```

Comment:
> ? means Something or Null ?

I am not sure but according to the other externs, it seems that an optional property would we described like this:
```
recent_outings: ?Array<Object>
```

Oh by the way: ``recent_outings`` and ``all_routes`` are not arrays! They look like:
```
recent_outings: {
  total: {number}
  routes: Array<Object>
}
```

Actually I was doing the same kind of externs in the ``association-tool-fix-2`` (not commited  yet, I'll probably have to use yours instead). It looks like
```
/**
 * @typedef {{
 *   associations: appx.Associations,
 *   orientations: ?Array.<string>
 * }}
 */
appx.DocumentData;


/**
 * @typedef {{
 *   images: Array.<Object>,
 *   users: ?Array.<Object>,
 *   routes: ?Array.<Object>,
 *   waypoints: ?Array.<Object>,
 *   waypoint_children: ?Array.<Object>,
 *   waypoint_parents: ?Array.<Object>,
 *   recent_outings: ?appx.RecentOutings,
 *   all_routes: ?appx.AllRoutes
 * }}
 */
appx.Associations;

/**
 * @typedef {{
 *   total: number,
 *   outings: Array.<Object>
 * }}
 */
appx.RecentOutings;

/**
 * @typedef {{
 *   total: number,
 *   routes: Array.<Object>
 * }}
 */
appx.AllRoutes;

```